Add support to disable downloading a runtime pack when RID is set

Currently the SDK provides a hook to disable downloading the targeting pack: `EnableTargetingPackDownload`, however if a `RuntimeIdentifier` is set it attempts to download a runtime pack even if that property is set to false.

This has caused trouble in dotnet/runtime where we build a targeting/runtime pack (as a directory not nupkg) and then we override the `PackageDirectory` to point to the corresponding directory, so we never need to download either the targeting or runtime pack. However, now that we already moved our target framework to use `net6.0` we had to add a `KnownTargetFramework` for `net6.0` and we set the `RuntimeFrameworkVersion` to `6.0.0`, since that runtime package doesn't exist in any of the feeds, we hit errors like:

```sh
/Users/santifdezm/repos/runtime/src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.browser-wasm with version (= 6.0.0)
/Users/santifdezm/repos/runtime/src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj : error NU1102:   - Found 437 version(s) in dotnet5 [ Nearest version: 6.0.0-alpha.1.20420.3 ]
/Users/santifdezm/repos/runtime/src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj : error NU1102:   - Found 396 version(s) in dotnet6 [ Nearest version: 6.0.0-alpha.1.20610.6 ]
/Users/santifdezm/repos/runtime/src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj : error NU1102:   - Found 8 version(s) in nuget.org [ Nearest version: 5.0.1 ]
/Users/santifdezm/repos/runtime/src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj : error NU1102:   - Found 0 version(s) in dotnet-eng
/Users/santifdezm/repos/runtime/src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj : error NU1102:   - Found 0 version(s) in dotnet-tools
/Users/santifdezm/repos/runtime/src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj : error NU1102:   - Found 0 version(s) in dotnet5-transport
/Users/santifdezm/repos/runtime/src/libraries/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj : error NU1102:   - Found 0 version(s) in dotnet6-transport
```

So we have had to add a workaround in a couple of places already to remove the package download with something like:
```xml
<Target Name="RemoveRuntimePackFromDownloadItem"
          Condition="'$(_UseLocalTargetingRuntimePack)' == 'true' and '$(RuntimeIdentifier)' != ''"
          AfterTargets="ProcessFrameworkReferences">
  <ItemGroup>
    <PackageDownload Remove="@(PackageDownload)"
                                       Condition="$([System.String]::Copy('%(Identity)').StartsWith('Microsoft.NETCore.App.Runtime'))" />
  </ItemGroup>
</Target>
```

It would be great if we could just set something like `EnableRuntimePackDownload=false` and this is skipped: 
https://github.com/dotnet/sdk/blob/8ed60c62b7fe25ecee657f79485325ebda970a84/src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs#L294-L309

cc: @dsplaisted @ViktorHofer @ericstj @Anipik

--- a/src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs
@@ -52,6 +52,8 @@ namespace Microsoft.NET.Build.Tasks
 
         public bool EnableTargetingPackDownload { get; set; }
 
+        public bool EnableRuntimePackDownload { get; set; }
+
         public ITaskItem[] FrameworkReferences { get; set; } = Array.Empty<ITaskItem>();
 
         public ITaskItem[] KnownFrameworkReferences { get; set; } = Array.Empty<ITaskItem>();
@@ -286,7 +288,7 @@ namespace Microsoft.NET.Build.Tasks
                     }
 
                     ProcessRuntimeIdentifier(hasRuntimePackAlwaysCopyLocal ? "any" : RuntimeIdentifier, runtimePackForRuntimeIDProcessing, runtimePackVersion, additionalFrameworkReferencesForRuntimePack,
-                        unrecognizedRuntimeIdentifiers, unavailableRuntimePacks, runtimePacks, packagesToDownload, isTrimmable, includeInPackageDownload);
+                        unrecognizedRuntimeIdentifiers, unavailableRuntimePacks, runtimePacks, packagesToDownload, isTrimmable, EnableRuntimePackDownload && includeInPackageDownload);
 
                     processedPrimaryRuntimeIdentifier = true;
                 }

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets
@@ -48,8 +48,9 @@ Copyright (c) .NET Foundation. All rights reserved.
       <FrameworkReference Include="@(_FrameworkReferenceToAdd)" />
     </ItemGroup>
 
-    <PropertyGroup Condition="'$(EnableTargetingPackDownload)' == ''">
-      <EnableTargetingPackDownload>true</EnableTargetingPackDownload>
+    <PropertyGroup>
+      <EnableTargetingPackDownload Condition="'$(EnableTargetingPackDownload)' == ''">true</EnableTargetingPackDownload>
+      <EnableRuntimePackDownload Condition="'$(EnableRuntimePackDownload)' == ''">true</EnableRuntimePackDownload>
     </PropertyGroup>
 
     <PropertyGroup>
@@ -78,6 +79,7 @@ Copyright (c) .NET Foundation. All rights reserved.
                                 TargetLatestRuntimePatch="$(TargetLatestRuntimePatch)"
                                 TargetLatestRuntimePatchIsDefault="$(_TargetLatestRuntimePatchIsDefault)"
                                 EnableTargetingPackDownload="$(EnableTargetingPackDownload)"
+                                EnableRuntimePackDownload="$(EnableRuntimePackDownload)"
                                 KnownCrossgen2Packs="@(KnownCrossgen2Pack)"
                                 NETCoreSdkRuntimeIdentifier="$(NETCoreSdkRuntimeIdentifier)">
 

